### PR TITLE
movement fixes

### LINKED ISSUE #8751: You can only move north-west as a ghost
First new movement bug yay

### Environment
- Game Version: 22060122
- OS Version: Win 10

### Detailed description of bug

you can only move in one direction as a ghost, whatever key you press.

### Reproduction steps
1. Become a ghost (spectating or aghosting)
2. Press any movement key
3. Ghost moves to north-west instead of desired direction

### LINKED ISSUE #8755: Can't click-move objects
Control-click an adjacent locker to start pulling it, then click on a tile that is adjacent to you and the locker. The locker should move there, but this no longer works.

### PR DESCRIPTION
**Pull request by Bod9001**
### Changelog:

fixes #8751
fixes #8755

CL: [Fix] Fixes ghost movement only working in two directions. #8751
CL: [Fix] You are not able to Pushing objects you are dragging  #8755
CL: [Fix]  error on start-ups fixed
CL: [Fix]  fixes being able to swap through window

### CODE REVIEW

**Comment by MaxIsJoe** [comment]
CL needs to be more descriptive 

`CL: [Fix] Fixes ghost movement only working in two directions.` is the only one i know what do the other two do

File: UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs [hunk 1]
```diff
@@ -367,11 +367,11 @@ public void StoreTo(ObjectContainer NewParent)
 
 
 	[ClientRpc]
-	public void RPCClientTilePush(Vector2Int WorldDirection, float speed, GameObject CausedByClient)
+	public void RPCClientTilePush(Vector2Int WorldDirection, float speed, GameObject CausedByClient, bool OverridePull)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs]:
```suggestion
	public void RPCClientTilePush(Vector2Int WorldDirection, float speed, GameObject CausedByClient, bool overridePull)
```

File: UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs [hunk 2]
```diff
@@ -367,11 +367,11 @@ public void StoreTo(ObjectContainer NewParent)
...
 	{
 		if (isServer) return;
 		if (PlayerManager.LocalPlayer == CausedByClient) return;
-		if (PulledBy.HasComponent) return;
+		if (PulledBy.HasComponent && OverridePull == false) return;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs]:
```suggestion
		if (PulledBy.HasComponent && overridePull == false) return;
```

File: UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs [hunk 3]
```diff
@@ -727,19 +730,19 @@ public bool CanPush(Vector2Int WorldDirection)
...
 	public void TryTilePush(Vector2Int WorldDirection, GameObject ByClient, float speed = Single.NaN,
-		UniversalObjectPhysics PushedBy = null)
+		UniversalObjectPhysics PushedBy = null, bool Overridepull =false)
 	{
 		if (isVisible == false) return;
 		if (PushedBy == this) return;
 		if (CanPush(WorldDirection))
 		{
-			ForceTilePush(WorldDirection, Pushing, ByClient, speed, PushedBy: PushedBy);
+			ForceTilePush(WorldDirection, Pushing, ByClient, speed, PushedBy: PushedBy, Overridepull: Overridepull);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs]:
```suggestion
			ForceTilePush(WorldDirection, Pushing, ByClient, speed, PushedBy: PushedBy, overridepull: overridepull);
```

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 4]
```diff
@@ -226,7 +226,7 @@ public void CmdSlideItem(Vector3Int destination)
 		var pushPull = playerScript.objectPhysics.Pulling.Component;
 		Vector3Int origin = pushPull.registerTile.WorldPositionServer;
 		Vector2Int dir = (Vector2Int)(destination - origin);
-		pushPull.TryTilePush(dir, null);
+		pushPull.TryTilePush(dir, null, Overridepull: true);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]:
```suggestion
		pushPull.TryTilePush(dir, null, overridepull: true);
```

**Review by corp-0 — COMMENTED** [review]
I'm on phone, please make sure those were all the usages of overridePull with the wrong case

**Review by corp-0 — APPROVED** [review]
I'm merging so we have it ready for tomorrow's playtest. I hope that log you left there isn't spamming that badly.

**Pull request merged**